db/src/db.rs

```diff
@@ -1015,14 +1015,29 @@ pub fn update_metadata(conn: &rusqlite::Connection, _old_schema: &Schema, new_sc
                      &[])?;
     }
 
+    // Populate the materialized view directly from datoms.
+    // It's possible that an "ident" was removed, along with its attributes.
+    // That's not counted as an "alteration" of attributes, so we explicitly check
+    // for non-emptiness of 'idents_altered'.
 
-    let mut stmt = conn.prepare(format!("INSERT INTO schema SELECT e, a, v, value_type_tag FROM datoms WHERE e = ? AND a IN {}", entids::SCHEMA_SQL_LIST.as_str()).as_str())?;
-    for &entid in &metadata_report.attributes_installed {
-        stmt.execute(&[&entid as &ToSql])?;
+    // TODO expand metadata report to allow for better signaling for the above.
+
+    if !metadata_report.attributes_installed.is_empty()
+        || !metadata_report.attributes_altered.is_empty()
+        || !metadata_report.idents_altered.is_empty() {
+
+        conn.execute(format!("DELETE FROM schema").as_str(),
```

Comment:
Maybe a little comment that we're using `:db/valueType` as a placeholder for the entire schema-defining set.